pkg/kubelet/cm/dra/state/state_checkpoint_test.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+	testutil "k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state/testing"
+)
+
+const testingCheckpoint = "dramanager_checkpoint_test"
+
+// assertStateEqual marks provided test as failed if provided states differ
+func assertStateEqual(t *testing.T, restoredState, expectedState ClaimInfoStateList) {
+	assert.Equal(t, expectedState, restoredState, "expected ClaimInfoState does not equal to restored one")
+}
+
+func TestCheckpointGetOrCreate(t *testing.T) {
+	testCases := []struct {
+		description       string
+		checkpointContent string
+		expectedError     string
+		expectedState     ClaimInfoStateList
+	}{
+		{
+			"Create non-existing checkpoint",
+			"",
+			"",
+			[]ClaimInfoState{},
+		},
+		{
+			"Restore valid checkpoint",
+			`{"version":"v1","entries":[{"DriverName":"dra.net.nvidia.com","ClaimUID":"067798be-454e-4be4-9047-1aa06aea63f7","ClaimName":"test-sf-claim-resource","Namespace":"default","PodUIDs":{"139cdb46-f989-4f17-9561-ca10cfb509a6":{}},"CdiDevices":["vendor.com/test=cdi-test"]}],"checksum":763324252}`,
+			"",
+			[]ClaimInfoState{{
+				DriverName: "dra.net.nvidia.com",
+				ClaimUID:   "067798be-454e-4be4-9047-1aa06aea63f7",
+				ClaimName:  "test-sf-claim-resource",
+				Namespace:  "default",
+				PodUIDs:    sets.New("139cdb46-f989-4f17-9561-ca10cfb509a6"),
+				CdiDevices: []string{"vendor.com/test=cdi-test"},
+			},
+			},
+		},
+		{
+			"Restore checkpoint with invalid checksum",
+			`{"version":"v1","entries":[{"DriverName":"dra.net.nvidia.com","ClaimUID":"067798be-454e-4be4-9047-1aa06aea63f7","ClaimName":"test-sf-claim-resource","Namespace":"default","PodUIDs":{"139cdb46-f989-4f17-9561-ca10cfb509a6":{}},"CdiDevices":["vendor.com/test=cdi-test"]}],"checksum":763324253}`,
```

Comment:
I just noticed all of these with `nvidia` in the name. We can't merge this like this. W need a more generic name.

Reply:
The useage of `vendor.com` is no good as well. I recommend just changing all references to these to `example.com`

Reply:
@klueska I'll lgtm it now and you can unhold when it's ready.